cmd/core/pfcp_connection.go

```diff
@@ -101,3 +113,50 @@ func (connection *PfcpConnection) SendMessage(msg message.Message, addr *net.UDP
 	}
 	return nil
 }
+
+// RefreshAssociations checks for expired associations and schedules heartbeats for those that are not expired.
+func (connection *PfcpConnection) RefreshAssociations() {
+	for assocAddr, assoc := range connection.NodeAssociations {
+		if assoc.IsExpired() {
+			log.Printf("Pruning expired node association: %s", assocAddr)
+			connection.DeleteAssociation(assocAddr)
```

Comment:
Are you sure that it's safe to delete association just in the middle of for cicle?

Reply:
This is a map and not an array/list so, it should be fine.
https://groups.google.com/g/golang-nuts/c/eJqwQLONhLs

> If map entries that have not yet been reached are deleted during iteration, the
corresponding iteration values will not be produced.